### PR TITLE
Attempt to automatically detect the current platform (monitored resource).

### PR DESCRIPTION
**Pull request by iantalarico**
Heavily based PR #735 by @chrisdunelm (thanks Chris :) )

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs]: This class (or something similar to it) should probably go in Google.Apis.Logging.v2
Then this code and the log4net code can share it.

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        /// <summary>The global resource.</summary>
+        public static readonly MonitoredResource GlobalResource = new MonitoredResource { Type = "global" };
+
+        /// <summary>
+        /// Attempts to detect the current <see cref="MonitoredResource"/> this is being run in.
+        /// If the monitored resource cannot be detected the global monitored resource is used.
+        /// </summary>
+        public static MonitoredResource DetectCurrentResource()
+        {
+            var platform = Platform.Instance();
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs]: Just to make sure you're aware that Platform.Instance() can block for up to 1 second the first time it's called.

Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs]: Yes, this is only ever called at start up.  So I *think* this is worthwhile.

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs (hunk 1 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs]: I think that would be a good idea, not point in duplicating the code.

My only worry is that (IIRC) MonitoredResource isn't a logging specific thing.  I think it's general for many Google APIs.

Comment (chrisdunelm) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs]: Yes, you're right.
It's actually in GAX ([MonitoredResource.cs](https://github.com/googleapis/gax-dotnet/blob/master/src/Google.Api.CommonProtos/Api/MonitoredResource.cs)), which means this common code can go there.
Perhaps leave this as it is for now, and file an issue to move this code into GAX, which I'll do soon.

**Comment by chrisdunelm** [comment]
Are the test failures anything serious?

**Comment by iantalarico** [comment]
They shouldn't be failing they are passing locally. Investigating now.

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs (hunk 1 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs]: Created an issue and cc'd you

**Comment by iantalarico** [comment]
Looks like auto detection was working :) 

Fixed tests.

**Pull request merged**